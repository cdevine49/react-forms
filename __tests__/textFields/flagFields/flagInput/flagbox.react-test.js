import React from 'react';
import Flagbox from '../../../../src/components/textFields/flagFields/flagInput/flagbox';
import renderer from 'react-test-renderer';

describe("Flagbox", () => {
  test("Without Props", () => {
    const tree = renderer.create(
      <Flagbox />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("CountryIndex", () => {
    let tree = renderer.create(
      <Flagbox countryIndex={5} />
    ).toJSON();

    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Flagbox countryIndex={10} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});