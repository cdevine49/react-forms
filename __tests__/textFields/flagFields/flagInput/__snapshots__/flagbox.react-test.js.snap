// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flagbox CountryIndex 1`] = `
<div
  className="flagbox-container"
>
  <button
    aria-pressed={false}
    className="flagbox"
    onClick={[Function]}
    type="button"
  >
    <div
      className="flag"
      style={
        Object {
          "backgroundImage": "url(/images/flags.png)",
          "backgroundPositionX": "0px",
          "backgroundPositionY": "-594px",
        }
      }
    />
    <img
      className="select-box-button"
      src="/images/select-box-button.png"
    />
  </button>
</div>
`;

exports[`Flagbox CountryIndex 2`] = `
<div
  className="flagbox-container"
>
  <button
    aria-pressed={false}
    className="flagbox"
    onClick={[Function]}
    type="button"
  >
    <div
      className="flag"
      style={
        Object {
          "backgroundImage": "url(/images/flags.png)",
          "backgroundPositionX": "0px",
          "backgroundPositionY": "-176px",
        }
      }
    />
    <img
      className="select-box-button"
      src="/images/select-box-button.png"
    />
  </button>
</div>
`;

exports[`Flagbox Without Props 1`] = `
<div
  className="flagbox-container"
>
  <button
    aria-pressed={false}
    className="flagbox"
    onClick={[Function]}
    type="button"
  >
    <div
      className="flag"
      style={
        Object {
          "backgroundImage": "url(/images/flags.png)",
          "backgroundPositionX": "0px",
          "backgroundPositionY": "-44px",
        }
      }
    />
    <img
      className="select-box-button"
      src="/images/select-box-button.png"
    />
  </button>
</div>
`;
